js/component.js

```diff
@@ -0,0 +1,25 @@
+/** @type {HTMLCanvasElement} */
+
+class Component {
```

Comment:
This class right here is going to be the class for the enemies, as you as passing in the constructor all this values:
`constructor(x, y, w, h, color, ctx)`. So far, everything correct. The only thing is that you needed to create another class for your player (or car in this case), where you don't need to pass the `color` in the constructor, and then pass the image as `new Image()`.